Z/z3/build_tarballs.jl

```diff
@@ -2,84 +2,54 @@
 # `julia build_tarballs.jl --help` to see a usage message.
 using BinaryBuilder
 
+julia_version = v"1.5.3"
 name = "z3"
-version = v"4.8.8"
+version = v"4.8.9"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/Z3Prover/z3.git", "ad55a1f1c617a7f0c3dd735c0780fc758424c7f1"),
-    ArchiveSource("https://julialang-s3.julialang.org/bin/linux/armv7l/1.3/julia-1.3.1-linux-armv7l.tar.gz", "965c8fab2214f8ce1b3d449d088561a6de61be42543b48c3bbadaed5b02bf824"; unpack_target="julia-arm-linux-gnueabihf"),
-    ArchiveSource("https://julialang-s3.julialang.org/bin/linux/x64/1.3/julia-1.3.1-linux-x86_64.tar.gz", "faa707c8343780a6fe5eaf13490355e8190acf8e2c189b9e7ecbddb0fa2643ad"; unpack_target="julia-x86_64-linux-gnu"),
-    ArchiveSource("https://github.com/Gnimuc/JuliaBuilder/releases/download/v1.3.0/julia-1.3.0-x86_64-apple-darwin14.tar.gz", "f2e5359f03314656c06e2a0a28a497f62e78f027dbe7f5155a5710b4914439b1"; unpack_target="julia-x86_64-apple-darwin14"),
-    ArchiveSource("https://github.com/Gnimuc/JuliaBuilder/releases/download/v1.3.0/julia-1.3.0-x86_64-w64-mingw32.tar.gz", "c7b2db68156150d0e882e98e39269301d7bf56660f4fc2e38ed2734a7a8d1551"; unpack_target="julia-x86_64-w64-mingw32"),
+    ArchiveSource("https://github.com/Z3Prover/z3/archive/z3-$(version).tar.gz",
+        "c9fd04b9b33be74fffaac3ec2bc2c320d1a4cc32e395203c55126b12a14ff3f4"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-case $target in
-  arm-linux-gnueabihf|x86_64-linux-gnu)
-    Julia_PREFIX=${WORKSPACE}/srcdir/julia-$target/julia-1.3.1
-    Julia_ARGS="-DZ3_BUILD_JULIA_BINDINGS=True -DJulia_PREFIX=${Julia_PREFIX}"
-    ;;
-  x86_64-apple-darwin14|x86_64-w64-mingw32)
-    Julia_PREFIX=${WORKSPACE}/srcdir/julia-$target/juliabin
-    Julia_ARGS="-DZ3_BUILD_JULIA_BINDINGS=True -DJulia_PREFIX=${Julia_PREFIX}"
-    ;;
-esac
-
-cd $WORKSPACE/srcdir/z3/
+cd z3*
 
 mkdir z3-build && cd z3-build
-cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
-    -DCMAKE_FIND_ROOT_PATH="${prefix}" \
+cmake \
     -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_FIND_ROOT_PATH="${prefix}" \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
-    ${Julia_ARGS} \
+    -DJulia_PREFIX="$prefix" \
+    -DZ3_BUILD_JULIA_BINDINGS=True \
     ..
```

Comment:
I took the liberty of sorting these arguments alphabetically and generally aligning them with other JLLs, in the hope that this will help a bit with future maintenance.